apps/backend/src/modules/points-of-interest/points-of-interest.service.ts

```diff
@@ -94,6 +97,33 @@ class PointsOfInterestService implements Service {
 		return item.toObject();
 	}
 
+	public async findPaginated(
+		options: PointsOfInterestPaginatedOptions,
+	): Promise<PointsOfInterestPaginatedResponseDto> {
+		const { page, perPage, search } = options;
+
+		const { items, total } =
+			await this.pointsOfInterestRepository.findPaginated({
+				page,
+				perPage,
+				search,
+			});
+
+		const totalPages = Math.ceil(total / perPage);
+
+		return {
+			data: items.map((item) =>
+				item.toObject(),
+			) as PointsOfInterestPaginatedSummary[],
```

Comment:
Do we need this cast? Shouldn't it automatically return the needed type?
```suggestion
			)
```

Reply:
yes,  I just wanted to do without toSummary methods Entity service and did this trick